content/docs/user-guide/programming/components/reflection/behavior-context.md

```diff
@@ -0,0 +1,461 @@
+---
+title: Behavior Context
+description: Use the behavior context to make runtime code accessible to scripting systems such as Script Canvas or Lua.
+weight: 300
+---
+
+The behavior context makes runtime code accessible to O3DE scripting systems such as Script Canvas or Lua. It provides script bindings that invoke runtime C++ methods, read constants, write properties, and create and handle EBus events. You can have multiple behavior contexts that are specialized for different purposes, and you can unreflect the behavior contexts to implement reloading.
+
+Use the behavior context to bind the following C++ constructs for scripting:
+
++ [Classes](#classes)
++ [Methods](#methods)
++ [Properties](#properties)
++ [Constants](#constants)
++ [Enums](#enums)
+
+In addition, the behavior context supports the O3DE [EBus](/docs/user-guide/programming/ebus) and [AZ::Event](/docs/user-guide/programming/az-event) event systems:
+
++ [EBus](#ebus)
+  + [Events](#ebus-events)
+  + [Event Handlers](#ebus-event-handlers)
+
+## Classes
+
+Reflects a C++ class or struct. You can provide an optional name. If you do not provide a name, the class name from `AzTypeInfo` is used. That name must be unique for the scope. Because the system uses `AzRTTI` to build the class hierarchy, you can use RTTI if you want to reflect base class functionality.
+
+Classes that are bound to the behavior context become objects that can be instantiated in a script environment. To reflect a class, you must provide the type that is reflected as a template argument to the class function.
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorcontext->Class<MyClass>();
+}
+```
+
+The base class should also be specified, if applicable:
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorcontext->Class<MyClass, TheBaseClass>();
+}
+```
+
+### Additional class information
```

Comment:
Provide a link to the API reference page which describes how to set these properties, or provide a brief example somewhere in this section about how to do the relevant overrides.

Reply:
Good idea. Will do...

Reply:
Added.